terraform/environments/data-platform-apps-and-tools/jml_secrets_manager_roles.tf

```diff
@@ -0,0 +1,44 @@
+# IAM role with a trust policy that allows the Lambda service to assume this role.
+resource "aws_iam_role" "secrets_manager_execution_role" {
```

Comment:
I'm not sure I understand why this has been made a separate policy? The lambda is not set up to do multiple role assumption in its current form. If there's a good reason (which I can believe there may be) we'll need to see adjustments to the lambda code to allow it to pick up different roles for the different actions.